ui-components/src/helpers/filters.ts

```diff
@@ -1,70 +1,217 @@
+import Listing from "@bloom-housing/backend-core/dist/src/listings/entities/listing.entity"
 
-function getComparisonForFilter(filterKey: ListingFilterKeys) {
-  switch (filterKey) {
-    case ListingFilterKeys.name:
-    case ListingFilterKeys.status:
-    case ListingFilterKeys.leasingAgents:
-      return EnumListingFilterParamsComparison["="]
-    case ListingFilterKeys.bedrooms:
-    case ListingFilterKeys.minRent:
-      return EnumListingFilterParamsComparison[">="]
-    case ListingFilterKeys.maxRent:
-      return EnumListingFilterParamsComparison["<="]
-    case ListingFilterKeys.zipcode:
-      return EnumListingFilterParamsComparison["IN"]
-    case ListingFilterKeys.seniorHousing:
-    case ListingFilterKeys.availability:
-    case ListingFilterKeys.ami:
-      return EnumListingFilterParamsComparison["NA"]
-    default: {
-      const _exhaustiveCheck: never = filterKey
-      return _exhaustiveCheck
+export const COMMUNITY_TYPE = "communityType"
+
+export const EMPTY_OPTION = { value: "", label: "" }
+
+// TODO: Select options should come from the database (#252)
+// The options need to be methods to avoid a race condition in translation
+// generation. The translations need to be generated after the app is
+// initialized.
+export const adaCompliantOptions: () => SelectOption[] = () => [
+  EMPTY_OPTION,
+  { value: "n", label: t("t.no") },
+  { value: "y", label: t("t.yes") },
+]
+
+export const preferredUnitOptions: () => SelectOption[] = () => [
+  EMPTY_OPTION,
+  { value: "0", label: t("listingFilters.bedroomsOptions.studioPlus") },
+  { value: "1", label: t("listingFilters.bedroomsOptions.onePlus") },
+  { value: "2", label: t("listingFilters.bedroomsOptions.twoPlus") },
+  { value: "3", label: t("listingFilters.bedroomsOptions.threePlus") },
+  { value: "4", label: t("listingFilters.bedroomsOptions.fourPlus") },
+]
+
+export const availabilityOptions: () => SelectOption[] = () => [
+  EMPTY_OPTION,
+  { value: AvailabilityFilterEnum.hasAvailability, label: t("listingFilters.hasAvailability") },
+  { value: AvailabilityFilterEnum.noAvailability, label: t("listingFilters.noAvailability") },
+  { value: AvailabilityFilterEnum.waitlist, label: t("listingFilters.waitlist") },
+]
+
+export const communityTypeOptions: () => SelectOption[] = () => [
+  EMPTY_OPTION,
+  { value: "seniorHousing", label: t("listingFilters.communityTypeOptions.senior") },
+  {
+    value: "specialNeeds",
+    label: t("listingFilters.communityTypeOptions.specialNeeds"),
+  },
+]
+
+/* Representation of the front end filters.
+ *
+ * This is decoupled from the backend representation in ListingFilterParams to
+ * allow greater flexibility in how filters are displayed in the front end.
+ */
+export class FrontEndFilter {
+  /* The frontend filter name. */
+  readonly name: string
+
+  /* The frontend filter value.
+   *
+   * Do not set this value directly! Use the setValue method in FrontEndFilters.
+   */
+  value: any
```

Comment:
But if I make it private here, I won't be able to access it in FrontEndFilters. Thoughts?

Reply:
Yes, sorry, I meant making the whole filters object on `FrontEndFilters` on line 93 private, so that these internal `FrontEndFilter` objects would only be accessible by the `FrontEndFilters` container object through getters/setters.

This suggestion seemed sort of out of scope for this PR so I didn't think it through all the way, but I know we'd have to replace a lot of the `filterState.filters[ListingFilterKeys.availability].value` with something like `filterState.getFilterValue(ListingFilterKeys.availability)`